### PR TITLE
Step 4

### PR DESCRIPTION
**Pull request by streichholz-ame**
add settings page functionality, fix authorisation issues

### CODE REVIEW

File: app/controllers/users/registrations_controller.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module Users
   class RegistrationsController < Devise::RegistrationsController
```
Comment (VanyaRuban) [inline app/controllers/users/registrations_controller.rb:2]: I think this file useless

File: app/decorators/account_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  delegate_all
+
+  # Define presentation-specific methods here. Helpers are accessed through
+  # `helpers` (aka `h`). You can override attributes, for example:
+  #
+  #   def created_at
+  #     helpers.content_tag :span, class: 'time' do
+  #       object.created_at.strftime("%a %m/%d/%y")
+  #     end
+  #   end
```
Comment (VanyaRuban) [inline app/decorators/account_decorator.rb:11]: please remove all commented code

File: app/helpers/address_helper.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module AddressHelper
```
Comment (VanyaRuban) [inline app/helpers/address_helper.rb:1]: please remove all useless files

File: app/views/addresses/_change_password.html.haml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+%p.in-gold-500.font-18.mb-25= t('account_settings.settings.password')
+= simple_form_for @current_user,  validate: true, defaults: { input_html: { class: 'general-form-md divider-lg-bottom pb-25' }}, url: account_path(current_user), html: { method: :put, id: "password" } do |f|
+  .form-group.mb-30
+    = f.input :old_password, placeholder: "Old Password", label: t('account_settings.settings.old_password'), label_html: { class: 'control-label input-label' }, input_html: { class: 'form-control', value: '' }
+  .form-group.mb-30
+    = f.input :new_password, placeholder: "New Password", label: t('account_settings.settings.new_password'), label_html: { class: 'control-label input-label' }, input_html: { class: 'form-control', value: '' }
+  .form-group.mb-40
+    = f.input :confirm_password, placeholder: "Confirm Password", label: t('account_settings.settings.confirm_password'), label_html: { class: 'control-label input-label' }, input_html: { class: 'form-control', value: '' }
```
Comment (VanyaRuban) [inline app/views/addresses/_change_password.html.haml:8]: placeholder I18n ? just suggestion

File: app/services/change_password_service.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
+class ChangePasswordService
+  attr_reader :current_user, :params
+
+  def initialize(current_user, params)
+    @current_user = current_user
+    @params = params
+  end
+
+  def call
+    change_password if passwords_valid(current_password)
```
Comment (VanyaRuban) [inline app/services/change_password_service.rb:10]: maybe rethink about `passwords_valid(current_password)`

**Review by VanyaRuban — CHANGES_REQUESTED** [review]

File: app/controllers/users/registrations_controller.rb (hunk 1 above)
Comment (streichholz-ame) [reply app/controllers/users/registrations_controller.rb:2]: cannot pass registration without this one

File: config/initializers/omniauth.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+OmniAuth.config.logger = Rails.logger
+
+Rails.application.config.middleware.use OmniAuth::Builder do
```
Comment (streichholz-ame) [inline config/initializers/omniauth.rb:3]: add key 'facebook'

File: app/presenters/address_presenter.rb [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+  end
+
+  def billing_form
+    AddressForm.new
+  end
+
+  def shipping_form
+    AddressForm.new
+  end
+
```
Comment (streichholz-ame) [inline app/presenters/address_presenter.rb:21]: private

**Pull request merged**